pkg/api/vlabs/validate.go

```diff
@@ -797,7 +797,7 @@ func (a *AgentPoolProfile) validateOrchestratorSpecificProperties(orchestratorTy
 func validateKeyVaultSecrets(secrets []KeyVaultSecrets, requireCertificateStore bool) error {
 	for _, s := range secrets {
 		if len(s.VaultCertificates) == 0 {
-			return fmt.Errorf("Invalid KeyVaultSecrets must have no empty VaultCertificates")
+			return fmt.Errorf("Valid KeyVaultSecrets must have no empty VaultCertificates")
```

Comment:
Also, Line 791 in validate.go is not really accessible via unit test. It is because of what happens in line 772. I think we can remove lines 791-793. If that's okay, I can update this PR

Reply:
Yes I think that makes sense, I'm all for removing unreachable code 👍 Your call whether you want to update this PR or leave it as a separate task